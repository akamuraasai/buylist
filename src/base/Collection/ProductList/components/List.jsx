import React from 'react';
import PropTypes from 'prop-types';

import Item from './Item';

const List = ({ items }) => (
  <div className="ProductList__list">
    {items.map(({ id, name, price, units, unitType, completed }) => (
      <Item
        key={id}
        name={name}
        price={price}
        units={units}
        unitType={unitType}
        completed={completed}
      />
    ))}
  </div>
);

List.defaultProps = {
  items: [],
};

List.propTypes = {
  items: PropTypes.arrayOf(PropTypes.shape({
    id: PropTypes.number,
    name: PropTypes.string,
    price: PropTypes.number,
    units: PropTypes.number,
    unitType: PropTypes.string,
    completed: PropTypes.bool,
  })),
};

export default List;
