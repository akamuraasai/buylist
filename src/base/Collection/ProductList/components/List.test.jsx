import React from 'react';
import { shallow } from 'enzyme';
import List from './List';

describe('ProductList > List', () => {
  it('should render without crashing', () => {
    const list = shallow(<List />);
    expect(list).toMatchSnapshot();
  });

  it('should render some items', () => {
    const props = {
      items: [
        { id: 1, unitType: 'kg', name: 'Tomate', price: 3.99, units: 1, completed: false },
        { id: 2, unitType: 'kg', name: 'Cebola', price: 1.99, units: 2.3, completed: false },
        { id: 3, unitType: 'kg', name: 'Alho', price: 4.99, units: 0.4, completed: true },
        { id: 4, unitType: 'un', name: 'Leite', price: 2.49, units: 12, completed: false },
      ],
    };

    const list = shallow(<List {...props} />);
    expect(list.find('Item').length).toEqual(4);
  })
});
