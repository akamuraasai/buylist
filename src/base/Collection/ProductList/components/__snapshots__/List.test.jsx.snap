// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductList > List should render some items 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <List
    items={
        Array [
            Object {
              "completed": false,
              "id": 1,
              "name": "Tomate",
              "price": 3.99,
              "unitType": "kg",
              "units": 1,
            },
            Object {
              "completed": false,
              "id": 2,
              "name": "Cebola",
              "price": 1.99,
              "unitType": "kg",
              "units": 2.3,
            },
            Object {
              "completed": true,
              "id": 3,
              "name": "Alho",
              "price": 4.99,
              "unitType": "kg",
              "units": 0.4,
            },
            Object {
              "completed": false,
              "id": 4,
              "name": "Leite",
              "price": 2.49,
              "unitType": "un",
              "units": 12,
            },
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Item
          completed={false}
          name="Tomate"
          price={3.99}
          unitType="kg"
          units={1}
/>,
        <Item
          completed={false}
          name="Cebola"
          price={1.99}
          unitType="kg"
          units={2.3}
/>,
        <Item
          completed={true}
          name="Alho"
          price={4.99}
          unitType="kg"
          units={0.4}
/>,
        <Item
          completed={false}
          name="Leite"
          price={2.49}
          unitType="un"
          units={12}
/>,
      ],
      "className": "ProductList__list",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "1",
        "nodeType": "function",
        "props": Object {
          "completed": false,
          "name": "Tomate",
          "price": 3.99,
          "unitType": "kg",
          "units": 1,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "2",
        "nodeType": "function",
        "props": Object {
          "completed": false,
          "name": "Cebola",
          "price": 1.99,
          "unitType": "kg",
          "units": 2.3,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "3",
        "nodeType": "function",
        "props": Object {
          "completed": true,
          "name": "Alho",
          "price": 4.99,
          "unitType": "kg",
          "units": 0.4,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "4",
        "nodeType": "function",
        "props": Object {
          "completed": false,
          "name": "Leite",
          "price": 2.49,
          "unitType": "un",
          "units": 12,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Item
            completed={false}
            name="Tomate"
            price={3.99}
            unitType="kg"
            units={1}
/>,
          <Item
            completed={false}
            name="Cebola"
            price={1.99}
            unitType="kg"
            units={2.3}
/>,
          <Item
            completed={true}
            name="Alho"
            price={4.99}
            unitType="kg"
            units={0.4}
/>,
          <Item
            completed={false}
            name="Leite"
            price={2.49}
            unitType="un"
            units={12}
/>,
        ],
        "className": "ProductList__list",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "function",
          "props": Object {
            "completed": false,
            "name": "Tomate",
            "price": 3.99,
            "unitType": "kg",
            "units": 1,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "2",
          "nodeType": "function",
          "props": Object {
            "completed": false,
            "name": "Cebola",
            "price": 1.99,
            "unitType": "kg",
            "units": 2.3,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "3",
          "nodeType": "function",
          "props": Object {
            "completed": true,
            "name": "Alho",
            "price": 4.99,
            "unitType": "kg",
            "units": 0.4,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "4",
          "nodeType": "function",
          "props": Object {
            "completed": false,
            "name": "Leite",
            "price": 2.49,
            "unitType": "un",
            "units": 12,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;

exports[`ProductList > List should render without crashing 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <List
    items={Array []}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [],
      "className": "ProductList__list",
    },
    "ref": null,
    "rendered": Array [],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [],
        "className": "ProductList__list",
      },
      "ref": null,
      "rendered": Array [],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
